app/controllers/articles_controller.rb

```diff
@@ -1,5 +1,6 @@
 class ArticlesController < ApplicationController
   before_action :load_article, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:index, :show]
```

Comment:
`before_action` move it on the top, after class declaration.
Why first load article if later may be you not authenticated?